native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java

```diff
@@ -104,6 +105,42 @@ protected void setXml(Connection connection, PreparedStatement preparedStatement
         }
     }
 
+    @Override
+    protected void setVarcharArray(Connection conn, PreparedStatement preparedStatement, int index,
+                                   Object value) throws DataError, SQLException {
+        Utils.throwUnsupportedArrayTypeError((BArray) value);
```

Comment:
Why basic array types are not supported? Varchar is string[] isn't it?

Reply:
The reason is that Oracle is only supporting varray types. In order to create varray we need to have element array and the varray type in oracledb. basic array types does not provide the type of the varray in sql as varray definition in sql as follows. 

```sql
CREATE OR REPLACE TYPE CharArrayType AS VARRAY(6) OF VARCHAR(100);
/
CREATE OR REPLACE TYPE IntArrayType AS VARRAY(6) OF NUMBER;
/

CREATE TABLE TestVarrayTable(
       PK               NUMBER GENERATED ALWAYS AS IDENTITY,
       COL_CHARARR      CharArrayType,
       COL_INTARR       IntArrayType,
       PRIMARY KEY(PK)
);
```